src/components/lib/cos_kernel_api.c

```diff
@@ -649,6 +649,10 @@ int
 cos_thd_switch(thdcap_t c)
 { return call_cap_op(c, 0, 0, 0, 0, 0); }
 
+int
+cos_thd_wakeup(thdcap_t thd, tcap_t tc, tcap_prio_t prio)
+{ return call_cap_op(tc, CAPTBL_OP_TCAP_WAKEUP, thd, (prio << 32) >> 32, prio >> 32, 0); }
```

Comment:
Well, we are talking about 2 shifts (assuming compiler doesn't do something tricky) versus 1 mask instruction (I believe both are close to free instructions in this case). Won't be a big difference but less instructions isn't bad ;)